sympy/core/add.py

```diff
@@ -140,7 +140,7 @@ def flatten(cls, seq):
                     # we know for sure the result will be nan
                     return [S.NaN], [], None
                 if coeff.is_Number:
-                    coeff += o
+                    coeff = o.__add__(coeff)
```

Comment:
Why is this change needed?

Reply:
So that adding a DecimalRational and a symbol doesn't move the DecimalRational back to a Rational.  See [this](https://github.com/meganly/sympy/blob/17648_DecimalRationalclass/sympy/core/tests/test_arit.py#L1878) test.

Reply:
Oh I see, it is o + coeff instead of coeff + o. It should work the same to use `+` here instead of `__add__`. 

I guess most of the stuff here has to do with getting this priority stuff to work right, so that when you combine DecimalRational and Rational it stays as DecimalRational.

Reply:
op_priority get the priority stuff right so this change isn't needed